docs/sources/developer/Developing-Ubuntu.md

```diff
@@ -0,0 +1,120 @@
+# Developing on Ubuntu
+
+## Git
+To download the latest version of Plaso you'll need to install the git tools:
+```
+sudo apt-get install git
+```
+
+Checkout the Plaso source from the git repo:
+```
+git clone https://github.com/log2timeline/plaso.git
+```
+
+## Dependencies
+
+Install the dependencies, using either:
+* A [virtualenv](Developing-Virtualenv.html#Fedora%20Core)
+* The [pre-packaged dependencies](Dependencies.html#Fedora%20Core).
+
+Check if you have all the dependencies installed and have the right minimum 
+versions:
+```
+python utils/check_dependencies.py
+```
+
+**Note that some dependencies are actively under development and can be 
+frequently updated, therefore we recommend checking the status of the
+ dependencies regularly.**
+
+## Updating your environment.
+
+To update the Plaso source code, run:
+```
+git pull origin master
+```
+
+If you are using a github fork your origin is pointing to your fork not, and not
+ the main Plaso git repository. When you run `git remote -v` You might see 
+ something like:
+```
+origin	https://github.com/Onager/plaso (fetch)
+origin	https://github.com/Onager/plaso (push)
+```
+
+Add a git remote called 'upstream' that you can use to sync your fork:
+```
+git remote add upstream https://github.com/log2timeline/plaso.git
+git pull --rebase upstream master
+```
+
+We provide packaged versions of the dependencies via the [
+l2tbinaries project](https://github.com/log2timeline/l2tbinaries/blob/master/README.md). 
+However it is possible that the dependencies are not fully up to date therefore we also provide a build script as part of [l2tdevtools project](https://github.com/log2timeline/l2tdevtools) to do unattended bulk builds.
+
+## Development tools
+If you intend to do development on Plaso you'll also need to install some 
+development tools:
+
+* PyLint
+* Python Mock
+
+### PyLint
+Currently plaso development uses PyLint version 1.9.x. 
+
+Remove any older version of PyLint.
+```bash
+sudo apt-get remove pylint
+```
+
+For pylint 1.9.x the following additional dependencies are required:
```

Comment:
describe installing this from the pylint3 GIFT PPA ?

Reply:
maybe remove the PyLint and Python Mock sections and replace this with an explanation about adding pylint3 PPA and installing pylint and mock via apt?

Reply:
I'd like to overhaul all these instructions, but I'll do that in a later change.

Reply:
ack